WooCommerce/Classes/ViewRelated/Themes/ThemesCarouselViewModel.swift

```diff
@@ -0,0 +1,111 @@
+import Foundation
+import Yosemite
+
+/// View model for `ThemesCarouselView`
+///
+final class ThemesCarouselViewModel: ObservableObject {
+
+    @Published private(set) var state: State = .loading
+    @Published private var themes: [WordPressTheme] = []
+    @Published private var currentThemeID: String?
+
+    let mode: Mode
+    private let stores: StoresManager
+
+    init(mode: Mode, stores: StoresManager = ServiceLocator.stores) {
+        self.mode = mode
+        self.stores = stores
+        observeThemeListAndCurrentTheme()
+    }
+
+    @MainActor
+    func fetchThemes() async {
+        state = .loading
+        do {
+            themes = try await loadSuggestedThemes()
+            /// Only update state immediately for the profiler flow.
+            /// The theme setting flow requires waiting for the current theme ID.
+            if mode == .storeCreationProfiler {
+                state = .content(themes: themes)
+            }
+        } catch {
+            DDLogError("⛔️ Error loading suggested themes: \(error)")
+            state = .error
+        }
+    }
+
+    func updateCurrentTheme(id: String?) {
+        currentThemeID = id
+    }
+}
+
+private extension ThemesCarouselViewModel {
+    func observeThemeListAndCurrentTheme() {
```

Comment:
Opinion: What do you think about renaming it to reflect the purpose of this method? I have `waitForCurrentThemeAndFinishLoading` in mind.

Reply:
Also handled this in a043a7a0a39868c00ca5b2092d108bd1655027e5.